Assets/Scripts/Management/UIManager.cs

```diff
@@ -0,0 +1,67 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using System;
+
+public class UIManager : MonoBehaviour
+{
+    [SerializeField]
+    private TextMeshProUGUI ScoreText;
+
+    private float elapsedTime;
+    private TimeSpan totalTime;
+    private bool isTimerRunning;
+
+    //create UIManager Instance for global referencing
+    public static UIManager Instance;
+
+    void Awake()
+    {
+        //Singleton logic
+        if (Instance == null)
+        {
+            Instance = this;
+        }
+        else
+        {
+            Destroy(this);
+            Debug.LogWarning($"Two instances of {nameof(UIManager)}, which is a Singleton, found in scene.");
+        }
+    }
+
+    private void Start()
+    {
+        isTimerRunning = false;
+        elapsedTime = 0f;
+    }
+
+    public void StartGameTimer()
+    {
+        ResetGameTimer();
+
+        isTimerRunning = true;
+        StartCoroutine(CountUpScore());
```

Comment:
This will start a coroutine every time the game resets, but does not discard the previous one if it is already running

Reply:
Also, the timer logic should probably me moved to the game management system, instead of the UI manager. But we can do that in a future PR.

Reply:
Added new commit to address coroutine bug and move logic to GameManager